Service Portal/Search Sources/DataFetchScript.js

```diff
@@ -0,0 +1,51 @@
+(function(query) {
+    var results = [];
+    var getuserdelegateof = new GlideRecord('sys_user_delegate');
+    getuserdelegateof.addQuery('delegate', gs.getUserID());
+	getuserdelegateof.addEncodedQuery("endsONToday@javascript:gs.beginningOfToday()@javascript:gs.endOfToday()");
+    getuserdelegateof.query();
+    if (getuserdelegateof.next()) {
```

Comment:
Your query could return multiple results, I'd suggest iterating through them and using a .hasNext() on this initial line